task/gimlet-inspector/src/main.rs

```diff
@@ -0,0 +1,140 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! The Gimlet Inspector provides a deliberately limited set of IPCs over the
+//! network, for extracting diagnostic data from a live system. This is intended
+//! to supplement the more general `dump_agent`.
+
+#![no_std]
+#![no_main]
+
+use drv_gimlet_seq_api::{SeqError, Sequencer};
+use gimlet_inspector_protocol::{
+    QueryV0, Request, SequencerRegistersResponseV0,
+};
+use hubpack::SerializedSize;
+use task_net_api::*;
+use userlib::*;
+
+task_slot!(NET, net);
+task_slot!(SEQ, seq);
+
+// Increase this if a new message requires a larger trailer.
+const MAX_REQ_TRAILER_SIZE: usize = 0;
+
+#[export_name = "main"]
+fn main() -> ! {
+    // Look up our peer task IDs and make clients.
+    let net = Net::from(NET.get_task_id());
+    let seq = Sequencer::from(SEQ.get_task_id());
+
+    const SOCKET: SocketName = SocketName::inspector;
+
+    loop {
+        // These buffers are currently kept kinda small because our protocol
+        // messages are small.
+        let mut rx_data_buf = [0u8; Request::MAX_SIZE + MAX_REQ_TRAILER_SIZE];
+        let mut tx_data_buf = [0u8; 128];
+
+        match net.recv_packet(
+            SOCKET,
+            LargePayloadBehavior::Discard,
+            &mut rx_data_buf,
+        ) {
+            Ok(mut meta) => {
+                let Ok((request, _trailer)) = hubpack::deserialize::<Request>(&rx_data_buf) else {
+                    // We ignore malformatted, truncated, etc. packets.
```

Comment:
It would be helpful to log into a ringbuf here, since we can read those remotely even on release builds using `dump-agent`

Reply:
I was really trying to keep ringbuf out of this task to keep the binary size and number of panic sites down. What about a counter? Counters can be read via dump agent too.

Reply:
Sure, a counter sounds good to me.

Reply:
Counters added, PTAL.